packages/integration-tests/tests/api/accounts.test.ts

```diff
@@ -2,104 +2,263 @@ import * as request from "supertest";
+        expect(typeof response.body.result[0].to).toStrictEqual("string");
+        expect(typeof response.body.result[0].value).toStrictEqual("string");
+        expect(typeof response.body.result[0].tokenName).toStrictEqual("string");
+        expect(typeof response.body.result[0].tokenSymbol).toStrictEqual("string");
+        expect(typeof response.body.result[0].tokenDecimal).toStrictEqual("string");
+        expect(typeof response.body.result[0].gas).toStrictEqual("string");
+        expect(typeof response.body.result[0].gasPrice).toStrictEqual("string");
+        expect(typeof response.body.result[0].input).toStrictEqual("string");
+        expect(typeof response.body.result[0].contractAddress).toStrictEqual("string");
+        expect(typeof response.body.result[0].cumulativeGasUsed).toStrictEqual("string");
+        expect(typeof response.body.result[0].contractAddress).toBeTruthy();
+        expect(typeof response.body.result[0].gasUsed).toStrictEqual("string");
+        expect(typeof response.body.result[0].confirmations).toStrictEqual("string");
+        expect(typeof response.body.result[0].fee).toStrictEqual("string");
+        expect(typeof response.body.result[0].l1BatchNumber).toBeTruthy();
+        expect(typeof response.body.result[0].transactionType).toStrictEqual("string");
+      });
+    });
+
+    //@id1806
+
+    it("Verify /api?module=account&action=tokennfttx", async () => {
+      await helper.retryTestAction(async () => {
+        const blocks = await request(environment.blockExplorerAPI).get("/blocks");
+        const blockNumber = blocks.body.items[0].number;
+        const nftAddress = await helper.getStringFromFile(bufferFile + Buffer.NFTtoL2);
+        console.log(nftAddress);
+        apiRoute = `/api?module=account&action=tokennfttx&page=1&offset=10&sort=desc&endblock${blockNumber}&startblock=0&contractaddress=${nftAddress}&address=${nftAddress}`;
+        response = await helper.performGETrequest(apiRoute);
+
+        expect(response.status).toBe(200);
+        expect(response.body).toStrictEqual(expect.objectContaining({ status: "1" }));
+        expect(response.body).toStrictEqual(expect.objectContaining({ message: "OK" }));
+        expect(typeof response.body.result[0].blockNumber).toStrictEqual("string");
+        expect(typeof response.body.result[0].timeStamp).toStrictEqual("string");
+        expect(typeof response.body.result[0].hash).toStrictEqual("string");
+        expect(typeof response.body.result[0].nonce).toStrictEqual("string");
+        expect(typeof response.body.result[0].blockHash).toStrictEqual("string");
+        expect(typeof response.body.result[0].transactionIndex).toStrictEqual("string");
+        expect(typeof response.body.result[0].from).toStrictEqual("string");
+        expect(typeof response.body.result[0].to).toStrictEqual("string");
+        expect(typeof response.body.result[0].tokenID).toStrictEqual("string");
```

Comment:
https://[block-explorer-api.testnets.zksync.dev/api?module=account&action=tokennfttx&page=1&offset=10&sort=desc&endblock=100&startblock=0&contractaddress=0x000000000000000000000000000000000000800A&address=0x000000000000000000000000000000000000800A](https://block-explorer-api.testnets.zksync.dev/api?module=account&action=tokennfttx&page=1&offset=10&sort=desc&endblock=100&startblock=0&contractaddress=0x000000000000000000000000000000000000800A&address=0x000000000000000000000000000000000000800A)

![image](https://github.com/matter-labs/block-explorer/assets/51121511/9e2ed916-2077-44da-89b5-f0289d962cd5)

what you will see in case not a nft address? I suppose ether current api data outdated or we have to use only correct contract address that belongs to NFT.

Reply:
It seems that this is will give us a case id1805

Reply:
Discussed this behaviour with Pavel and Roman - it was decided, that this behaviour is intended. 
Based on the fact that checking whether address belongs to NFT is complicated and time-consuming.